SearchEngine/lambda_SearchEngine/include/CnDictionary.h

```diff
@@ -0,0 +1,31 @@
+#ifndef __CNDICTIONARY_H__
+#define __CNDICTIONARY_H__
+
+#include "Dictionary.h"
+
+
+class CnDictionary
+:public Dictionary
+{
+public:
+    CnDictionary(const string & dictpath);
+    ~CnDictionary();
+    void init();
+
+    pair<string,int> getDict(size_t num);
+    set<int> getIndex(const string &);
+
+    void show(const string &);
+private:
+    vector<pair<string,int>> _dict;
```

Comment:
it is better to change a pair to a custome define struct.
since the latter one is more semantic clearer.

Reply:
should keep naming styple in mind.
 _dictpath, should be dict_path_